### ISSUE TITLE
detect C++20 flags and add automatically

### ISSUE THREAD

**Issue by jeffhammond** [triSYCL/triSYCL#284]
You are using C++20, which recent releases of Clang and GCC should partially support.

Surely there is some recipe for C++20 flag detection and application using CMake already...

```
In file included from /Users/jrhammon/Work/SYCL/triSYCL/tests/jacobi/include/jacobi-stencil.hpp:4:
/Users/jrhammon/Work/SYCL/triSYCL/tests/jacobi/include/stencil-fxd.hpp:248:15: warning: explicit capture of 'this' with a capture default of '=' is a C++20 extension
      [-Wc++20-extensions]
          [=, this] (cl::sycl::id<2> id) {
```

**Comment by keryell** [comment]
Yes, this is the annoying part because I was using this code before without `this` and I had some warnings with C++20.
So, I added it to avoid warnings with C++20 but now I have some warnings when not using C++20... :-(
Anyway, we are in 2020 right? :-)

**Comment by jeffhammond** [comment]
> Anyway, we are in 2020 right? :-)

I would argue we are actually all dead and in Hell, but I don't mind calling it 2020 🤷‍♂️
